python/pyarrow/tests/test_scalars.py

```diff
@@ -102,6 +100,37 @@ def test_basics(value, ty, klass, pickle_module):
     assert wr() is None
 
 
+@pytest.mark.numpy
+def test_basics_np_required(pickle_module):
+    value, ty, klass = np.float16(1.0), pa.float16(), pa.HalfFloatScalar
+    s = pa.scalar(value, type=ty)
```

Comment:
What we are testing here essentially is that a numpy float scalar (`value`) can be converted to a pyarrow Scalar with `pa.scalar(..)`. But all the rest of the test is not specific to that fact that we created it from a numpy scalar, so I don't think you need to repeat that entirely here (the test above already covers all those things for HalfFloatScalar), so I think you can simplify the test here.

Reply:
And also, it seems this conversion from numpy scalar to pyarrow scalar is also covered in the test below (`test_numerics`) ?

Reply:
I am not sure I understand how the test above is testing `HalfFloatScalar` if I remove from the parametrization:
`(np.float16(1.0), pa.float16(), pa.HalfFloatScalar),`
I am trying to remove `np` from parametrize in order to avoid collection failures.
@jorisvandenbossche are you suggesting I can just remove `(np.float16(1.0), pa.float16(), pa.HalfFloatScalar),` as this one is already covered on the rest of the tests and avoid adding a new test for this case?

Reply:
> I am not sure I understand how the test above is testing `HalfFloatScalar` if I remove from the parametrization:
> `(np.float16(1.0), pa.float16(), pa.HalfFloatScalar),`

Ah, sorry, I assumed that it would still be in the paramatrization, as we are also testing the others without a numpy scalar. Is there a reason the test above cannot use `(1.0, pa.float16(), pa.HalfFloatScalar)` for the parametrization?

Reply:
And to answer my own question: because we cannot create a half float scalar right now from a plain python float:

```
In [7]: pa.scalar(1.0, pa.float16())
...
ArrowTypeError: Expected np.float16 instance
```

OK, you can ignore my comment here ;)

Reply:
Can you maybe just add a comment at the beginning of `test_basics_np_required` explaining the context (it's duplicating `test_basics` but for float16, which requires a numpy scalar to create it)

Reply:
Thanks @jorisvandenbossche I've added a comment